### ISSUE TITLE
BadgedResource tooltip on resource name

### ISSUE THREAD

**Issue by lizagilman** [patternfly/patternfly-react#1238]
Hi,
Using `BadgedResource` component I would like to have a tooltip on the `resourceName` prop. AFAIU, a tooltip is only shown on the badge itself, but not on the `resourceName`. 

If I pass a DOM fragment (in my case, a `resourceName` with overlay and react router link) instead of an expected `string` it solves my problem, but then I get proptypes validation warning.

Considering [implementation](https://github.com/patternfly/patternfly-react/blob/3e1d440f1900b9a55e7a928492c8c7b9b00536d2/packages/patternfly-3/patternfly-react-extensions/src/components/BadgedResource/BadgedResource.js#L72) , can we not restrict to a `string`? Or provide node as `children` ?

My use case:

Initial goal - have a resource badge with link and tootlip on `resourceName`

code: 
```
    <BadgedResource
          resourceName={clusterName}
          kindStr="Cluster"
          kindAbbr="CL"
          badgeColor="#00b9e4"
      />
```

where `clusterName` is -
```  
    <OverlayTrigger 
        overlay={<Tooltip id={cluster.id}>{``cluster name: ${cluster.name}``}</Tooltip>}
        placement="right"
      >
        <Link to={``/cluster/${cluster.id}``}>{name.trim() !== '' ? name : cluster.name}</Link>
      </OverlayTrigger>
    );
```
Looks like works well, but `resourceName` expected to be a `string`. 

@jeff-phillips-18 @priley86  what do you guys think?

/cc @cben  @Laviro @sharvit

**Comment by sharvit** [comment]
It makes sense to me that `resourceName` should actually be the `children` while the children should be from the type of `node`.

Together with your example, I see it getting used as:
```jsx
<OverlayTrigger 
  overlay={<Tooltip id={cluster.id}>{``cluster name: ${cluster.name}``}</Tooltip>}
  placement="right"
>
  <BadgedResource
    kindStr="Cluster"
    kindAbbr="CL"
    badgeColor="#00b9e4"
  >
    <Link to={``/cluster/${cluster.id}``}>{name.trim() !== '' ? name : cluster.name}</Link>
  </BadgedResource>
</OverlayTrigger>
```

Btw, you might be able to use it with the current implementation as:
```jsx
<OverlayTrigger 
  overlay={<Tooltip id={cluster.id}>{``cluster name: ${cluster.name}``}</Tooltip>}
  placement="right"
>
  <Link to={``/cluster/${cluster.id}``}>
    <BadgedResource
      resourceName={name.trim() !== '' ? name : cluster.name}
      kindStr="Cluster"
      kindAbbr="CL"
      badgeColor="#00b9e4"
    />
  </Link>
</OverlayTrigger>
```

**Comment by lizagilman** [comment]
@sharvit thanks !

It does work, but as the tooltip is applied on the whole `BadgedResource` component, the tooltip location is far away from actual `resourceName`, as the `BadgedResource` width is much greated then the acutual `resourceName`

**Comment by jeff-phillips-18** [comment]
I'd be in favor of making the  `resourceName` property accept node or string.

Handling children would be ok as well, but we'd have to continue support of the `resourceName` prop so as to not break existing uses.

**Comment by lizagilman** [comment]
AFAIK node would accept anything that can be rendered, including string

**Comment by jeff-phillips-18** [comment]
Fixed by #1260
